locallib.php

```diff
@@ -106,27 +106,89 @@ function hvp_get_core_assets($context) {
     $settings['loadedJs'] = array();
     $settings['loadedCss'] = array();
 
-    // Make sure files are reloaded for each plugin update.
-    $cachebuster = \hvp_get_cache_buster();
-
     // Use relative URL to support both http and https.
     $liburl = \mod_hvp\view_assets::getsiteroot() . '/mod/hvp/library/';
     $relpath = '/' . preg_replace('/^[^:]+:\/\/[^\/]+\//', '', $liburl);
 
     // Add core stylesheets.
     foreach (\H5PCore::$styles as $style) {
-        $settings['core']['styles'][] = $relpath . $style . $cachebuster;
-        $PAGE->requires->css(new moodle_url($liburl . $style . $cachebuster));
+        $url = generate_css_url('library/' . $style);
+        $settings['core']['styles'][] = $url->out(false);
+        $PAGE->requires->css($url);
     }
     // Add core JavaScript.
     foreach (\H5PCore::$scripts as $script) {
-        $settings['core']['scripts'][] = $relpath . $script . $cachebuster;
-        $PAGE->requires->js(new moodle_url($liburl . $script . $cachebuster), true);
+        $scriptpath = $relpath . $script;
+        $settings['core']['scripts'][] = generate_js_url($scriptpath)->out(false);
+        $PAGE->requires->js($scriptpath, true);
     }
 
     return $settings;
 }
 
+/**
+ * Determine the correct JS Revision to use for this load.
+ * Copied from outputrequirementslib because that function is protected.
+ *
+ * @return int the jsrev to use.
+ */
+function get_jsrev(): int {
+    global $CFG;
+
+    if (empty($CFG->cachejs)) {
+        $jsrev = -1;
+    } else if (empty($CFG->jsrev)) {
+        $jsrev = 1;
+    } else {
+        $jsrev = $CFG->jsrev;
+    }
+
+    return $jsrev;
+}
+
+/**
+ * Generates a cacheable URL to serve a JS file.
+ * This is used for URLS that will be used inside javascript. For inside Moodle code,
+ * $PAGE->requires->js() will perform the needed conversion.
+ *
+ * @param string $scriptpath
+ * @return moodle_url
+ */
+function generate_js_url(string $scriptpath): moodle_url {
+    global $CFG;
+
+    $jsrev = get_jsrev();
+
+    if (empty($CFG->slasharguments)) {
```

Comment:
Do we really care about slashargument support either way? Why not make it consistent and just always use params?

Reply:
I can do that. I just copied what I found elsewhere in the code.